src/System.Text.Json/tests/JsonNodeTests.JsonNumber.cs

```diff
@@ -0,0 +1,632 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Text.Json.Tests
+{
+    public static partial class JsonNodeTests
```

Comment:
I think I'd make this just JsonNumberTests, not partial on JsonNodeTests.  JsonNumber is testable in its own right, and you might want to just run all the JsonNumber tests (e..g `dotnet msbuild /t:BuildAndTest /p:XunitOptions="-class System.Text.Json.Tests.JsonNumberTests"`)